src/Http/Requests/FileListingRequest.php

```diff
@@ -22,6 +23,31 @@ public function authorize()
     public function rules()
     {
         return [
+            'directory' => 'required',
+            'page' => 'nullable|integer|min:1',
+            'perPage' => 'nullable|integer|min:0',
         ];
     }
+
+    public function getDirectory(): string
+    {
+        return $this->get('directory');
+    }
+
+    public function getPage(): int
+    {
+        $page = $this->get('page');
+        return $page == null ? 1 : $page;
+    }
+
+    public function getPerPage(): int
+    {
+        $perPage = $this->get('perPage');
+        return $perPage == null ? 50 : $perPage;
```

Comment:
You will get the same thing if you use the default value: `return $this->get('perPage', 50);`